### PR TITLE
MM-20387: Migrate 'components/sidebar_right' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
This PR migrates 'components/sidebar_right' module and associated tests to TypeScript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16719
JIRA: https://mattermost.atlassian.net/browse/MM-20387

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/sidebar_right/sidebar_right.tsx [hunk 1]
```diff
@@ -211,11 +223,7 @@ export default class SidebarRight extends React.PureComponent {
             content = (
                 <div className='post-right__container'>
                     <FileUploadOverlay overlayType='right'/>
-                    <RhsThread
-                        previousRhsState={previousRhsState}
-                        toggleSize={this.toggleSize}
-                        shrink={this.onShrink}
```
Comment (sinansonmez) [inline components/sidebar_right/sidebar_right.tsx:217]: I see these props are not requested / used in `RhsThread` props, so I removed them

File: components/sidebar_right/sidebar_right.tsx [hunk 2]
```diff
@@ -14,40 +13,51 @@ import RhsThread from 'components/rhs_thread';
...
-            closeRightHandSide: PropTypes.func.isRequired,
-            openAtPrevious: PropTypes.func.isRequired,
-            updateSearchTerms: PropTypes.func.isRequired,
-            showChannelFiles: PropTypes.func.isRequired,
-            showChannelInfo: PropTypes.func.isRequired,
-        }),
+import {Channel} from '@mattermost/types/channels';
+import {RhsState} from 'types/store/rhs';
+
+type Props = {
```
Comment (michelengelen) [inline components/sidebar_right/sidebar_right.tsx:23]: not exactly sure what is the current approach to this, but I like to have the typings in the `index.ts` file and separate it into `OwnProps`, `StateProps` and `Actions`. Then import it into the component file and combine them to a `Props` type. 

```
type Props = OwnProps & StateProps & Actions;
```

WDYT @hmhealey ?

File: components/sidebar_right/sidebar_right.tsx [hunk 3]
```diff
@@ -180,12 +190,14 @@ export default class SidebarRight extends React.PureComponent {
         this.props.actions.setRhsExpanded(false);
     };
 
-    handleUpdateSearchTerms = (term) => {
+    handleUpdateSearchTerms = (term: string) => {
         this.props.actions.updateSearchTerms(term);
-        this.focusSearchBar();
+        if (this.focusSearchBar) {
+            this.focusSearchBar();
+        }
```
Comment (michelengelen) [inline components/sidebar_right/sidebar_right.tsx:197]:
```suggestion
        this.focusSearchBar?.();
```

**Review by michelengelen — APPROVED** [review]
I have only one suggestion and a more generalistic comment about type placement/structure that is unrelated to the changes here and will probably need further discussion anyways

File: components/sidebar_right/sidebar_right.tsx (hunk 2 above)
Comment (sinansonmez) [reply components/sidebar_right/sidebar_right.tsx:23]: Please let me know about decision. Based on the decision, I can use the suggested approach in future PRs

**Review by AshishDhama — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @sinansonmez 
Tested, looks good to merge.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/218362-b1e7ee7-onprem-ent-webapp-pr-10726-mattermost/mm-ee-testb1e7ee7/mochawesome.html) - no PR related failures.

**Pull request merged**